src/Transform/ONNX/InstrumentONNXPass.cpp

```diff
@@ -88,32 +88,34 @@ class InstrumentONNXPass
     if (instrumentONNXOps == "" || instrumentONNXOps == "NONE")
       return;
     init(instrumentONNXOps);
-    auto function = getFunction();
-    auto &funcBody = function.getBody();
 
-    // Iterate on the operations
-    for (Operation &op : funcBody.getOps()) {
-      if (isa<mlir::ONNXOpsDialect>(op.getDialect())) {
+    // Iterate on the operations nested in this function
+    getFunction().walk([&](mlir::Operation *op) {
+      if (isa<mlir::ONNXOpsDialect>(op->getDialect())) {
         // Skip the prefix "onnx." of onnx op name
-        const char *opName = op.getName().getStringRef().data() + 5;
+        const char *opName = op->getName().getStringRef().data() + 5;
         if (!allOpsAllowed && allowedOps.find(opName) == allowedOps.end())
-          continue;
+          return;
 
-        Location loc = op.getLoc();
-        OpBuilder opBuilder(&op);
+        // Do not instrument ONNXReturn, which has terminate Trait
+        if (strcmp(opName, "Return") == 0)
```

Comment:
Suggest to use strncmp.

Reply:
Would it be not better to  use an isa<xxxReturn> instead of comparing a name?

Reply:
That's true. I will use isa.

Reply:
Actually, my new version checks the Trait instead of the Op. This method can handle any op that has to be at the end of block.

Reply:
Even better